app/states/unified_sqlmodels.py

```
@@ -0,0 +1,1467 @@
+                
+                if address_data.get("state"):
+                    conditions.append("state = :state")
+                    params["state"] = address_data["state"]
+                
+                if address_data.get("zip_code"):
+                    conditions.append("zip_code = :zip_code")
+                    params["zip_code"] = address_data["zip_code"]
+                
+                # Need at least street_1 and city to find a match
+                if len(conditions) >= 2:
+                    query = f"SELECT * FROM unified_addresses WHERE {' AND '.join(conditions)} LIMIT 1"
```

Ruff (0.14.7)

Possible SQL injection vector through string-based query construction

(S608)